keyring/backends/OS_X.py

```diff
@@ -41,17 +44,40 @@ def set_password(self, service, username, password):
             raise PasswordSetError("Can't store password on keychain: " "{}".format(e))
 
     def get_password(self, service, username):
-        if username is None:
-            username = ''
+        creds={}
+        if (not username):
+            prompt = "find-generic-password"
+            
+            if (service.find(".com")>=0):
+                prompt = "find-internet-password"
+            
+            output = execute(
+                    'security 2>&1 '+prompt+' -g -s '+service,
+                    lambda x: "",
+                    lambda x: "",
+                    ignore_exit_codes=True
+                )
 
-        try:
-            return api.find_generic_password(self.keychain, service, username)
-        except api.NotFound:
-            pass
-        except api.KeychainDenied as e:
-            raise KeyringLocked("Can't get password from keychain: " "{}".format(e))
-        except api.Error as e:
-            raise KeyringError("Can't get password from keychain: " "{}".format(e))
+            find_passwd = re.compile('password: "([^"]+)"').search
+            find_user = re.compile('"acct"<blob>="([^"]+)"').search
+            creds['username'] = find_key(find_user, output)
+            creds['password'] = find_key(find_passwd, output)
+
+        else:
+            try:
+                if (service.find(".com")>=0):
```

Comment:
I'm pretty sure `.com` isn't a suitable indicator of whether a service represents an internet address.

Reply:
I agree, that was a highly non sufficient solution. I'll instead check for https or http